src/main/java/com/cloudbees/jenkins/support/impl/SmartLogFetcher.java

```diff
@@ -106,28 +106,16 @@ public ForNode forNode(Node n) throws IOException {
                     final long localLength = local.length();
                     if (entry.getValue() < Long.MAX_VALUE) {
                         // only copy the new content
-                        FileOutputStream fos = null;
-                        InputStream is = null;
-                        try {
-                            fos = new FileOutputStream(local, true);
-                            is = remoteDir.child(entry.getKey()).readFromOffset(localLength);
+                        try (FileOutputStream fos = new FileOutputStream(local, true);
+                             InputStream is = remoteDir.child(entry.getKey()).readFromOffset(localLength)) {
```

Comment:
Is the _close_ the same as _closeQuietly_? or may the close throw an exception?

Reply:
Good catch. The behaviour is different now. When `close` of an `AutoClosable` throws an exception, this is visible now.

The question is: Was the former implementation, hiding exceptions, the better one.

See https://github.com/jenkinsci/jenkins/pull/2830 and [IO-504](https://issues.apache.org/jira/browse/IO-504) for similar changes.